### ISSUE TITLE
"required field missing" error when start docker-compose with deck v1.11.0

### ISSUE THREAD

**Issue by xibuka** [Kong/deck#605]
deck version
v1.11.0

Running `. ./start docker-compose` using `se-tools` with all the features disabled,  I saw the below error message, and the script stopped at there.
```
...
creating plugin response-transformer for route 6b71ab11-f299-46e4-b5b4-8a927d79b0df
creating plugin key-auth for service b18fe6d6-cc7c-4dcf-8401-10ce7e4a7500
creating plugin rate-limiting-advanced for route 263b8300-1a19-4537-ae58-c4387fc8dca1 and consumer 688f93a3-4ab9-4a4b-aa9d-89769b64322d
Summary:
  Created: 15
  Updated: 43
  Deleted: 0
Error: 1 errors occurred:
	while processing event: {Create} plugin rate-limiting-advanced for route 263b8300-1a19-4537-ae58-c4387fc8dca1 and consumer 688f93a3-4ab9-4a4b-aa9d-89769b64322d failed: HTTP status 400 (message: "schema violation (config.namespace: required field missing)")
```

And I also hit another error with the following message when enabling some of the features. 
```
...
creating route ldap-session
creating plugin openid-connect for route bbac09c7-7a42-4c54-b228-a86543d41d5d
creating plugin oauth2 for service 4d463f5b-233b-40ed-8219-e707d8307408
Summary:
  Created: 132
  Updated: 0
  Deleted: 0
Error: 1 errors occurred:
	while processing event: {Create} plugin oauth2 for service 4d463f5b-233b-40ed-8219-e707d8307408 failed: HTTP status 400 (message: "schema violation (config.provision_key: required field missing)")

```

It looks like some of the settings are missed in the YAML file when using deck `1.11.0`.
After switching to deck `1.10.0`, the above issues are no longer shown, and the environment is building successfully.

**Comment by GGabriele** [comment]
Hello @xibuka ,

what's a sample content of your config file? Are you using the plugin config dedup feature? I suspect this may be related to [this other issue](https://github.com/Kong/deck/issues/593), which we recently fixed.

**Comment by xibuka** [comment]
Hi @GGabriele 

I'm not using a config file manually but using the below tools to automate the setup of the demo environment. 
https://github.com/Kong/se-tools/tree/main/demo-environment

There is a [featuresPicker.sh](https://github.com/Kong/se-tools/blob/main/demo-environment/featuresPicker.sh) tool for generating the config YAML file, and the issue part is as below. 

```
- name: gold-silver-free
  service:
    name: httpbin
  paths:
  - /goldSilverFree
  strip_path: true
  plugins:
  - name: key-auth
    config:
      key_names:
      - apikey
      anonymous: fa2a6660-6fa5-4bec-b695-e8ceae9a038d
  - name: rate-limiting-advanced
    config:
      limit:
      - 3
      sync_rate: 10
      window_size:
      - 60
```

When deck v1.11.0 was installed, it will complain that there is no namespace in the config section. 
```
(config.namespace: required field missing)
```

And when using deck v1.10.0 instead, there is no issue at all and the demo environment is created successfully.

**Comment by GGabriele** [comment]
Hi @xibuka ,

I see what's the problem here. We will work on this!

**Comment by GGabriele** [comment]
Hi @xibuka ,

tl;dr: users should include `auto` fields in their configuration file

As you can see from the discussion on [this related PR](https://github.com/Kong/go-kong/pull/152), we are going to not fix this on purpose: this is currently failing because the `namespace` field is an `auto` field, meaning that if the user doesn't provide any input, Kong autogenerates it. Since `deck` is a tool for declarative configuration, we believe this is a good place for `deck` to fail since not having a `namespace` and autogenerate it here would mean that decK is not pushing configuration declaratively: a side-effect of this would be that consecutive deployments with a configuration file not including any `namespace` would result in misleading diffs, since Kong will keep regenerating a `namespace` each time.

I'm going to close this now. Please feel free to comment back or open a new one if you have any other questions or doubts!
